### ISSUE TITLE
Setup uiowa07

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#7017]
# Story

As a web team, we should try setting up uiowa07 application so that we can verify if a Cloud Next application will work with our setup.

# Additional Info

>Yes - the new application has been provisioned but we were waiting on confirmation that your current load balancers would be associated with the new application (since it's spun up in a new pod).

- Access has been granted to the appropriate teams.

# Proposed solution

- [x] Grant appropriate teams permissions to this application
- [x] Follow docs here https://iowa.sharepoint.com/sites/drupal-developer-documentation/SitePages/Acquia%20Cloud/Acquia%20Cloud.aspx
- [x] Any specific testing we want to try to help us understand if Cloud Next will work with our setup or what needs to change

```[tasklist]
### Tasks
- [x] Ensure RA's Update process is set to "Do not inform".
- [x] PHP settings
- [x] New Relic app ID
- [x] Enable production mode on prod
- [x] New Relic variables on prod
- [x] Enable Platform Email from configuration for each environment.
- [x] Add slack webhook variable to all ENV
- [x] Ensure that teams have the necessary access to the new application - working session for ddev?
- [ ] Enable Log Forwarding
- [x] Contact it-security about log forwarding
- [x] Scheduled task for cron and prod site counts.
- [x] Create and install a new SSL certificate in production.
- [x] Added secrets.settings.php to each environment
- [x] Create release
- [x] Switch the code branch to develop-build for develop, main-build for stage, and after a release the build tag for prod.
- [x] Install the default site on each environment
- [x] Delete the WELCOME tag that a new application ships with.
- [x] Confirm cron job (via blt) works once a site has been installed.
- [x] Any other testing the team feels would be valuable
- [x] Authentication on data.its
- [x] Slack notifications for new relic
- [x] git deploy sync for main-build
- [x] git deploy sync for develop-build
```

**Comment by briand44** [comment]
Added blocked label until we hear about load balancers. I think that could be a blocker to using our current domain setup/strategy and possibly other issues.

**Comment by briand44** [comment]
> We have successfully completed the change to your load balancers from shared to dedicated. Dedicated balancers were added to prod, dev/stg environments ([bal-15037](https://cci.acquia.com/server/details/nojs/prod/bal-15037), [bal-15038](https://cci.acquia.com/server/details/nojs/prod/bal-15038)). Before removing shared, please update all domains with all environments. You will have 5 days, to repoint DNS to the new IP address.
> 
> IP Address for new Dedicated Load Balancers: *54.163.225.50 (5459)*.
> 
> Please let me know if you have any questions.

https://acquia.my.site.com/s/case/5006g00000xyThkAAE/repoint-dns-uiowa07

**Comment by joewhitsitt** [comment]
Grooming convo: Started with medium based on previous app setups and "upsized" it to large to buffer for unknowns.

**Comment by joewhitsitt** [comment]
>Make sure that a release was done recently and that main is clear of recent merges.

Going to do some of the tasks, but should wait till the next release to finish with the git operations.

**Comment by joewhitsitt** [comment]
>Enable Platform Email from configuration for each environment.

I am no longer seeing per environment config here. uiowa.edu domain has been associated with uiowa07 platform email.

**Comment by joewhitsitt** [comment]
@briand44 i may need assistance with the application's permission settings. i think there is something special you can do that I can't do with regard to this.

**Comment by joewhitsitt** [comment]
Getting an `An unknown error occurred. Please try your request again. If the problem persists, contact Acquia Support.` when trying to configure log forwarding

**Comment by joewhitsitt** [comment]
Going to open a support ticket about a couple things. EIPs, Log Forwarding, Scheduled job logging path.

**Comment by joewhitsitt** [comment]
Not certain yet, but I think the following needs to be updated from:

```
/var/www/html/${AH_SITE_NAME}/vendor/bin/blt ume cron -v --no-interaction &>> /var/log/sites/${AH_SITE_NAME}/logs/$(hostname -s)/drush-cron.log
```

to 

```
/var/www/html/${AH_SITE_NAME}/vendor/bin/blt ume cron -v --no-interaction &>> /shared/logs/sites/${AH_SITE_NAME}/logs/$(hostname -s)/drush-cron.log
```

https://docs.acquia.com/cloud-platform/faq-acn/#how-does-logging-behave-differently-in-cloud-next

> Cloud Next has a new directory for cron logs and custom logs that must be routinely pruned to prevent unnecessary growth of your file system

**Comment by joewhitsitt** [comment]
Continued the cron offset at 7 min, but optimistic that we can move away from the offset with Cloud Next applications.

**Comment by joewhitsitt** [comment]
As mentioned here, https://iowaweb.slack.com/archives/CSQA18BK4/p1701296738558039?thread_ts=1701296460.641989&cid=CSQA18BK4, I was wondering if we should provision/install/test a website before adding a custom website to this app.

Test cron, site counts, hawkid authentication, facebook oembed (secrets), config/updb deployment (artifact:update:drupal:all-sites)

I don't think the default site part of the `ume` processes (cron, site counts, etc.)

**Comment by joewhitsitt** [comment]
Got permission from some of the team to provision a sandboxnext site on uiowa07 with the intention of testing the deprovision command at the end of everything.

**Comment by joewhitsitt** [comment]
Test ideas

- [x] Provision/installation
- [ ] ~~SSL coverage~~ Going to assume this will be fine to avoid a CNAME request to hostmaster
- [x] Authentication
- [x] Cron
- [x] oEmbed secrets (facebook)
- [x] Thank you program (blocked until Web EIPs are established)
- [x] Confirm splunk data (blocked until log forwarding is established)
- [x] Config change import/updb (perhaps leave it for 10.1 deployment)?
- [x] sql-sync/rsync
- [x] Cross application sync? (simulate domain change)
- [ ] Deprovision (blocked until all other test steps have been completed)

**Comment by joewhitsitt** [comment]
test provision https://github.com/uiowa/uiowa/pull/7153

**Comment by joewhitsitt** [comment]
for some reason the stage slack webhook isn't firing. it did for dev https://iowaweb.slack.com/archives/CSQA18BK4/p1701284901636279

i wonder if Rick's slack code is flexible enough to work with the different naming convention? https://github.com/uiowa/uiowa/pull/7148/files

**Comment by joewhitsitt** [comment]
Running into this: https://acquia.my.site.com/s/article/360046548293-Troubleshooting-The-provided-host-name-is-not-valid-for-this-server-error-message

sitesitenownext.prod.drupal.uiowa.edu

opened a separate ticket as this seems different

**Comment by joewhitsitt** [comment]
I forgot to suggest in planning that I think this should move to blocked/waiting. I don't think I can proceed with the remaining issue summary items or the "test site" items without first hearing back from Acquia.

Needless to say, support should refrain from provisioning new websites on uiowa07 until this issue is complete.

**Comment by joewhitsitt** [comment]
Looks like part of this became unblocked over the weekend. I will respond to one of the tickets. Not sure if they did anything or it just took time to resolve.

**Comment by joewhitsitt** [comment]
A request for Web EIPs has been initiated so I am told. Seeking clarification on whether or not that will resolve the log forwarding error we get in the UI.

I think I fixed the cron job by adjusting the log path: 

Cloud Classic environments:

/var/log/sites/example.prod/logs/web-9876/drush-cron.log.

Cloud Next environments:

/shared/logs/drush-cron.log

**Comment by joewhitsitt** [comment]
Facebook oembed worked



![Image](https://github.com/uiowa/uiowa/assets/4663676/df275731-d79c-490c-8b76-302d52a5e226)

**Comment by joewhitsitt** [comment]
Going to simulate a cross-application domain change. Copying sandbox.prod over.

update: ✅ db/files/private-files all seem to sync fine

**Comment by joewhitsitt** [comment]
I think this can move back to blocked/waiting as we continue to wait for next steps from Acquia regarding Web EIPs and Log Forwarding. Deprovision _should_ work based on what I am seeing but we will want a test site for the previous items.

**Comment by joewhitsitt** [comment]
Web EIPs were setup

Just sent a thank you to @briand44 from the uiowa07 website. It said it was successful. Was it?

Or wait, do we need to tell HR the Web EIPs first?

**Comment by briand44** [comment]
Yeah, I got a thank you and a thank you (copy). Not sure why this worked though? Maybe because its behind our load balancer?

**Comment by joewhitsitt** [comment]
Shoot. Maybe we should have tested it without the Web EIPs in place...

**Comment by briand44** [comment]
They are not allowlisted so it shouldn't matter that they are in place. Did you check the supervisor option?

**Comment by joewhitsitt** [comment]
I didn't check the supervisor option

**Comment by briand44** [comment]
can you test with that checked?

**Comment by joewhitsitt** [comment]
Done

**Comment by briand44** [comment]
same result, I received both emails

**Comment by joewhitsitt** [comment]
For log forwarding, Acquia enabled it (no error trying to load the config form), but there is a 401 timeout in trying to install splunk with the same config as another app. There is an email out to Nikki/it-security as well as a response to the Acquia ticket about this.

**Comment by joewhitsitt** [comment]
>The D9 Thank You form in sitenow_dispatch is using data.its.uiowa.edu instead of querying the HR API directly

@briand44 we should have looked at the code earlier.. 

https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/sitenow_dispatch/src/Form/ThankYouForm.php#L112

It does indeed reach out to data.its to get the hr data instead, so I assume the comment about removing Access Control List (ACL) rules would mean to tell HR folks to remove the IPs other than data.its now that oneit.uiowa.edu has launched. Was that the only site using the thank you form before on v1?

Huge if true, because the only other service I am aware of would be the CleryEdge on safety.uiowa.edu right? If Acquia charges for Web EIPs, we could hand some back.

**Comment by joewhitsitt** [comment]
Shouldn't have spoken too soon. Web EIPs might be the glue needed for log forwarding on ACN. Passed the newly created ones along to ISPO so they can review their firewall rules.

**Comment by briand44** [comment]
>Passed the newly created ones along to ISPO so they can review their firewall rules.

Did you document this on https://iowa.sharepoint.com/sites/drupal-developer-documentation/SitePages/Miscellaneous/On-premise%20server%20ACL/On-premise%20server%20ACL.aspx?

Did you just pass along the prod EIPs?

If we need to allowlist EIPs for log forwarding or other things, it would be good to capture in our docs because there will be alot of IPs to allowlist compared to the handful currently.

**Comment by joewhitsitt** [comment]
>Did you just pass along the prod EIPs?

Yes, I figured we would debug with uiowa07, but I also thought we were only forwarding logs for PROD. Looks like where it has been setup, they are all failing.

**Comment by briand44** [comment]
>I also thought we were only forwarding logs for PROD

That was my understanding as well

**Comment by joewhitsitt** [comment]
Sent a followup email to ISPO today. Acquia said they are going to hold on the current ticket until the firewall rules are reviewed.

**Comment by joewhitsitt** [comment]
ISPO had setup an initial set of rules yesterday to try. No luck, however the firewall logs indicated some more rules might be needed. They are going to work on that.

Closed the Acquia ticket for now since this seems to be an issue on our side of things.

**Comment by joewhitsitt** [comment]
Just had a chat with @briand44 about this. The log forwarding is still "not quite right" but we are seeing application activity on splunk. `index=uiowa-acquia app=uiowa07prod`

Brian responded to the Acquia ticket, but we think the deprovision of this test site can happen. I'll see if I can get to that in a little bit

**Comment by joewhitsitt** [comment]
Fudge... looks like the `blt umd` command has an issue with the file deletion and doesn't finish the rest.

Seem to be getting a permission denied for public and private file directories. Also noticed a separate directory that has dashes instead of periods and contains the private files directory:

<img width="534" alt="Screenshot 2024-01-03 at 1 04 02 PM" src="https://github.com/uiowa/uiowa/assets/4663676/b9c79ee2-4f64-4653-b575-66773d90c73d">

**Comment by joewhitsitt** [comment]
Will come back to this.

**Comment by joewhitsitt** [comment]
Noticing a difference in how the permissions are setup:

```
drwxr-x--- uiowa02 uiowa02 sandbox.uiowa.edu
---
drwxr-x--- uiowa02 uiowa02 files
drwx------ uiowa02 uiowa02 files-private 

drwxr-xr-x. root cloudadmin sandboxnext.sites.uiowa.edu
---
drwxr-xr-x. uiowa07.dev uiowa07 files
drwxr-xr-x. uiowa07.dev uiowa07 files-private
```

```
* Note that we CD into the file system first and THEN delete the site files
* directory. If we just rm -rf the directory and $site is ever empty, the
* entire sites directory would be deleted.
```

My understanding is that the script checks the site directory to see if it is empty before deleting the site directory, but since it is owned by root and the group is cloudadmin, we don't have the permissions to delete it.

So either we switch to a "gut it, but leave it" method or we need to open a ticket with Acquia about how to remove the site directory.

**Comment by joewhitsitt** [comment]
Opened up a ticket with Acquia about this ^^

**Comment by joewhitsitt** [comment]
Seeking some clarification about this, but:

>We recommend `rm -rf https://sandboxnext.sites.uiowa.edu/files/*` (the /* is important), should work on both Cloud Next and Classic.

We will need to adjust and run for both files and files-private

**Comment by joewhitsitt** [comment]
Got past dev files with https://github.com/uiowa/uiowa/pull/7246/commits/3ff5259d88f2089202585b53fd288775467c5c6f . Now running into the same "test" "stage" terminology issue as before with setup.

**Comment by briand44** [comment]
Created a separate bug for the log forwarding issue. https://github.com/uiowa/uiowa/issues/7284. I think this can be closed now.

**Comment by briand44** [comment]
@joewhitsitt can we start putting sites on this application?

**Comment by joewhitsitt** [comment]
If you don't think the log forwarding is a blocker, i'd say so. I didn't try all of our commands against this application (e.g. transfer site between apps), but so far we have found solutions to all the bumps so far.
